packages/ckeditor5-table/src/plaintableoutput.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module table/plaintableoutput
+ */
+
+import { Plugin } from 'ckeditor5/src/core';
+import Table from './table';
+import TableCaption from './tablecaption';
+import TablePropertiesEditing from './tableproperties/tablepropertiesediting';
+
+/**
+ * The plain table output feature.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class PlainTableOutput extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'PlainTableOutput';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ Table, TablePropertiesEditing, TableCaption ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+
+		// Override default table data downcast converter.
+		editor.conversion.for( 'dataDowncast' ).elementToStructure( {
+			model: {
+				name: 'table',
+				attributes: [ 'headingRows' ]
+			},
+			view: downcastTableElement,
+			converterPriority: 'high'
+		} );
+
+		// Make sure <caption> is always downcasted to <caption> in the data pipeline.
+		editor.conversion.for( 'dataDowncast' ).elementToElement( {
+			model: 'caption',
+			view: 'caption',
+			converterPriority: 'high'
+		} );
+
+		// Handle border-style, border-color, border-width and background-color table attributes.
+		downcastTableBorderAndBackgroundAttributes( editor );
+	}
+}
+
+// The plain table downcast converter callback.
+//
+// @private
+// @param {module:engine/model/element~Element} Table model element.
+// @param {module:engine/conversion/downcastdispatcher~DowncastConversionApi} conversionApi The conversion API object.
+// @returns {module:engine/view/containerelement~ContainerElement} Created element.
+function downcastTableElement( modelElement, { writer } ) {
+	// Table body rows slot.
+	const rowsSlot = writer.createSlot( element => element.is( 'element', 'tableRow' ) );
```

Comment:
Is it expected that we don't use thead?

Reply:
Yes. It is expected.

The markup for "plain" table looks like this:

```html
<table>
  <caption>Monthly savings</caption>
  <tr>
    <th>Month</th>
    <th>Savings</th>
  </tr>
  <tr>
    <td>January</td>
    <td>$100</td>
  </tr>
</table>
````
https://www.w3schools.com/tags/tag_caption.asp

There are no `<thead>` not `<tbody>` tags in it nor they should be. The `<tbody>` is actually added automatically by the browser so it won't hurt to add it on our end.

Header rows are just a normal rows with `<th>` cells inside `<tbody>`.

Reply:
Well, I understand it in a different way, as per [ticket](https://github.com/ckeditor/ckeditor5/issues/10892) description:

> The goal is to mimic CKE4's output.

1. Go here: https://ckeditor.com/docs/ckeditor4/latest/examples/table.html
2. Go to the source editing
3. Paste the table code from my previous comment.
4. Go back to editor
5. Check the source editing again

You get:

```html
<table>
	<caption>Monthly savings</caption>
	<tbody>
		<tr>
			<th>Month</th>
			<th>Savings</th>
		</tr>
		<tr>
			<td>January</td>
			<td>$100</td>
		</tr>
	</tbody>
</table>
```

That's all I care about to meet requirements.

Reply:
CKEditor 4's table plugin will output `<thead>`:

1.  Insert table (via the toolbar button)
2.  It indeed has only `<tbody>`
3.  Open table properties
4.  Set "Table headers" to "First row"
5.  Check out the source again:

```html
<table border="1" cellpadding="1" cellspacing="1" style="width:500px">
	<thead>
		<tr>
			<th scope="col">&nbsp;</th>
			<th scope="col">&nbsp;</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>&nbsp;</td>
			<td>&nbsp;</td>
		</tr>
		<tr>
			<td>&nbsp;</td>
			<td>&nbsp;</td>
		</tr>
	</tbody>
</table>
```

Reply:
The goal is to mimic CKE4 output, as you mentioned, and CKE4 will use `<thead>` if necessary.

Reply:
> if necessary

How do I determine when it's necessary and when not?

Reply:
I can't distinguish between the example I posted, and the example you posted @Reinmar.

So I just changed in this way: if there are some heading rows in the model, it will create `<thead>`. If not, it won't.

This means it will work as per your example.

In case someone paste in source editing (or sets the data with `setData()`):

```html
<table>
	<caption>Monthly savings</caption>
	<tbody>
		<tr>
			<th>Month</th>
			<th>Savings</th>
		</tr>
		<tr>
			<td>January</td>
			<td>$100</td>
		</tr>
	</tbody>
</table>
```

The output will be:

```html
<table>
	<caption>Monthly savings</caption>
	<thead>
		<tr>
			<th>Month</th>
			<th>Savings</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>January</td>
			<td>$100</td>
		</tr>
	</tbody>
</table>
```

So it will be affected.